Tasks/GoToolV0/gotool.ts

```diff
@@ -119,20 +116,173 @@ function setGoEnvironmentVariables(goRoot: string) {
     }
 }
 
-// This function is required to convert the version 1.10 to 1.10.0.
-// Because caching utility accept only sementic version,
-// which have patch number as well.
-function fixVersion(version: string): string {
-    let versionPart = version.split(".");
-    if(versionPart[1] == null) {
-        //append minor and patch version if not available
-        return version.concat(".0.0");
-    }
-    else if(versionPart[2] == null) {
-        //append patch version if not available
-        return version.concat(".0");
-    } 
-    return version;
+function isSemverWithPatch(version: string): boolean {
+    // Accepts versions of the form major.minor.patch or major.minor.patch-revision
+    return /^\d+\.\d+\.\d+(?:-\d+)?$/.test(version.trim());
```

Comment:
The name of this function doesn't really mean what I think it would mean anymore, and it makes it harder to understand `resolveVersionAndCache` than I think it needs to be... I think it might be best to not try to share this particular function, and lean more towards using parsed versions in `resolveVersionAndCache` so a function like this isn't necessary.